test/alternator/test_describe_table.py

```diff
@@ -57,20 +57,56 @@ def test_describe_table_billing(test_table):
     assert 'LastUpdateToPayPerRequestDateTime' in got['BillingModeSummary']
     assert got['BillingModeSummary']['LastUpdateToPayPerRequestDateTime'] == got['CreationDateTime']
 
-# Test that DescribeTable correctly returns the table's creation time.
-# We don't know what this creation time is supposed to be, so this test
-# cannot be very thorough... We currently just tests against something we
-# know to be wrong - returning the *current* time, which changes on every
-# call.
-@pytest.mark.xfail(reason="DescribeTable does not return table creation time")
-def test_describe_table_creation_time(test_table):
-    got = test_table.meta.client.describe_table(TableName=test_table.name)['Table']
-    assert 'CreationDateTime' in got
-    time1 = got['CreationDateTime']
-    time.sleep(1) 
-    got = test_table.meta.client.describe_table(TableName=test_table.name)['Table']
-    time2 = got['CreationDateTime']
-    assert time1 == time2
+# Test that DescribeTable correctly returns the table's creation time - 
+# timestamp in seconds as double with as much precision as the operating system will provide
+def test_describe_table_creation_time(dynamodb):
+    before_time = time.time()
+    
+    table1 = create_test_table(dynamodb,
+        KeySchema=[ { 'AttributeName': 'p', 'KeyType': 'HASH' },
+                    { 'AttributeName': 'c', 'KeyType': 'RANGE' }
+        ],
+        AttributeDefinitions=[
+                    { 'AttributeName': 'p', 'AttributeType': 'S' },
+                    { 'AttributeName': 'c', 'AttributeType': 'S' },
+        ])
+    # let's sleep few ms, so table2 creation time was always bigger, as we now return CreationDateTime in ms precision
+    time.sleep(0.002)    
+    table2 = create_test_table(dynamodb,
+        KeySchema=[ { 'AttributeName': 'p', 'KeyType': 'HASH' },
+                    { 'AttributeName': 'c', 'KeyType': 'RANGE' }
+        ],
+        AttributeDefinitions=[
+                    { 'AttributeName': 'p', 'AttributeType': 'S' },
+                    { 'AttributeName': 'c', 'AttributeType': 'S' },
+        ])
+
+    # we'll read tables in reverse in case code is just returning now() time
+    got2 = table2.meta.client.describe_table(TableName=table2.name)['Table']
+    got1 = table1.meta.client.describe_table(TableName=table1.name)['Table']
+    assert 'CreationDateTime' in got1
+    assert 'CreationDateTime' in got2
+    
+    time1 = got1['CreationDateTime']
+    time2 = got2['CreationDateTime']
+    
+    after_time = time.time()
+    # we add some wiggle room, as the test might be run with on aws and time synchronization might be bit off
+    assert before_time - 1  < time1.timestamp() < time2.timestamp() < after_time + 1, (before_time, time1, time1.timestamp(), time2, time2.timestamp(), after_time)
```

Comment:
Wow, I never knew it is legal in Python to do something like a < b < c !!!

Why do you need that parathesized stuff at the end? Do you feel it helps you understand error cases? Isn't pytest's excellent assert rewriting good enough?